secretsmanager/src/base64.cpp

```diff
@@ -20,50 +20,21 @@
 
 namespace concord::secretsmanager {
 
-const std::string salt_prefix{"Salted__"};
-const uint8_t salt_size = 8;
-
-std::string base64Enc(const std::vector<uint8_t>& salt, const std::vector<uint8_t>& cipher_text) {
-  if (salt.size() != salt_size) {
-    throw std::runtime_error("Bad salt size " + std::to_string(salt.size()) + ". Expected value " +
-                             std::to_string(salt_size));
-  }
-
+std::string base64Enc(const std::vector<uint8_t>& cipher_text) {
   CryptoPP::Base64Encoder encoder;
-  encoder.Put((unsigned char*)salt_prefix.data(), salt_prefix.length());
-  encoder.Put(salt.data(), salt.size());
   encoder.Put(cipher_text.data(), cipher_text.size());
   encoder.MessageEnd();
-
   uint64_t output_size = encoder.MaxRetrievable();
   std::string output(output_size, '0');
   encoder.Get((unsigned char*)output.data(), output.size());
 
   return output;
 }
 
-SaltedCipher base64Dec(const std::string& input) {
-  std::vector<uint8_t> decoded;
-  CryptoPP::StringSource ss(input, true, new CryptoPP::Base64Decoder(new CryptoPP::VectorSink(decoded)));
-
-  SaltedCipher result;
+std::vector<uint8_t> base64Dec(const std::string& input) {
+  std::vector<uint8_t> dec;
+  CryptoPP::StringSource ss(input, true, new CryptoPP::Base64Decoder(new CryptoPP::VectorSink(dec)));
```

Comment:
How is the decoder destroyed after the explicit new?
Does `StringSource` take ownership of the `Base64Decoder?`?

Reply:
This is one of the weirdnesses of crypto++. Their [README](https://github.com/weidai11/cryptopp/blob/b72007386d0fdb817d95a299f7368646ffd50afd/Readme.txt#L105) states this:

> 1. If a constructor for A takes a pointer to an object B (except primitive
> types such as int and char), then A owns B and will delete B at A's
> destruction.  If a constructor for A takes a reference to an object B,
> then the caller retains ownership of B and should not destroy it until
> A no longer needs it.

So yes - StringSource takes ownership and releases it. It was confusing for me too.

Reply:
Thanks for the info !